pyEpiabm/pyEpiabm/tests/test_sweep/test_host_progression_sweep.py

```diff
@@ -176,6 +189,116 @@ def test_icdf_exception_raise(self):
+        try:
+            # Stocks the real time steps per day value
+            real = pe.Parameters.instance().time_steps_per_day
+
+            # Assigns temporarily a new value for time steps per day to test
+            # multiple time steps per day case
+            pe.Parameters.instance().time_steps_per_day = 100
+            # Same test as test_infectiousness_progression:
+            infectious_period = pe.Parameters.instance().asympt_infect_period
+            model_time_step = 1 / pe.Parameters.instance().time_steps_per_day
+            num_infectious_ts = int(np.ceil(infectious_period /
+                                            model_time_step))
+            test_sweep = pe.sweep.HostProgressionSweep()
+            infect_prog = test_sweep.infectiousness_progression
+            # Checks output type is numpy array
+            self.assertIsInstance(infect_prog, np.ndarray)
+            # Checks elements are 0 after k and greater than 0 before
+            tail = infect_prog[num_infectious_ts:2550]
+            zeros = np.zeros(2550-num_infectious_ts)
+            self.assertTrue((tail == zeros).all())
+            self.assertTrue((infect_prog[0:num_infectious_ts] >
+                            np.zeros(num_infectious_ts)).all())
+
+            # Assigns temporarily a new value for time steps per day to raise
+            # error
+            pe.Parameters.instance().time_steps_per_day = 10000
+            with self.assertRaises(ValueError):
+                pe.sweep.HostProgressionSweep()
+
+            # Resets the value of time steps per day
+            pe.Parameters.instance().time_steps_per_day = real
+        except Exception as e:
+            raise e
+
+    def test_limit_case_infectiousness_progression(self):
+        """Tests the case in the infectiousness progression method where the
+        parameter j would be greater or equal to the infectiousness profile
+        resolution. In that case, the elements of the array infectiousness
+        progression are simply set to 0.
+        """
+        num_infectious_ts =\
+            int(np.ceil(pe.Parameters.instance().asympt_infect_period
+                        * pe.Parameters.instance().time_steps_per_day))
+        with mock.patch('numpy.floor') as mock_floor:
+            # We need to mock the np.floor function to have j greater or equal
+            # to the infectiousness profile resolution
+            mock_floor.return_value = 57
+            test_sweep = pe.sweep.HostProgressionSweep()
+            infect_prog = test_sweep.infectiousness_progression
+            # Checks that the output is a numpy array
+            self.assertIsInstance(infect_prog, np.ndarray)
+            # Checks that all elements are equal to 0
+            zeros = np.zeros(2550)
+            self.assertTrue((infect_prog == zeros).all())
+            # Checks that it is called the right number of times
+            self.assertEqual(mock_floor.call_count, num_infectious_ts + 1)
```

Comment:
Needs a comment to explain the +1 and why its okay (i.e. that it is also called in the init function, but for an attribute that isn't used here)/ Also worth checking with @rcw5890 if there's a better way to do this!